paper-autocomplete-chips.js

```diff
@@ -365,27 +379,33 @@ class PaperAutocompleteChips extends translatable(PolymerElement) {
 	 */
 	_clearItemSelection(event) {
 		const item = event.model.item,
-			selectedIndex = this.selectedItems.indexOf(item);
-
+			tempIndex = this.selectedItems.indexOf(item.description),
+			selectedIndex = tempIndex !== -1 ? tempIndex : this.selectedItems.indexOf(item);
 		// This will remove from the DOM the source element of the processed event ...
 		this.splice('selectedItems', selectedIndex, 1);
 		// ... so we must prevent further propagation of this event, because its source is now invalid.
 		// (This has caused troubles in app-drawer-layout click event handler).
 		event.preventDefault();
 		event.stopPropagation();
 	}
-	/**
-	 * Get text for the chip labels.
-	 * @param {object} item Item.
-	 * @param {string} textProperty Property in the item where the text is located.
-	 * @returns {void}
-	 */
-	_getChipText(item, textProperty) {
-		let localItem = item;
-		if (typeof localItem === 'object') {
-			localItem = this.get(textProperty, localItem) || localItem.text;
+	enrichChips(itemList, textProperty) {
+		if (itemList == null || !Array.isArray(itemList.base)) {
+			return [];
 		}
-		return localItem !== '' ? localItem : this.noValueLabel;
+		return itemList.base.map(chip => {
+			if (typeof chip === 'object') {
+				const localDescription = this.get(textProperty, chip) || chip.text;
+				chip.hidden = !localDescription;
+				chip.description = localDescription;
+				chip.title = localDescription !== '' ? localDescription : this.noValueLabel;
+				return chip;
```

Comment:
```suggestion
				return {
						...chip,
						hidden: !localDescription,
						// etc
				};
```

ES6 destructuring approach. This will also make sure that we don't edit existing objects but create new ones for the chip-purpose.